ethdb/privateapi/ethbackend.go

```diff
@@ -54,10 +58,13 @@ type EthBackendServer struct {
 	reverseDownloadCh chan<- PayloadMessage
 	// Notify whether the current block being processed is Valid or not
 	statusCh <-chan ExecutionStatus
-	// Last block number sent over via reverseDownloadCh
-	numberSent uint64
 	// Determines whether stageloop is processing a block or not
-	waitingForBeaconChain *uint32 // atomic boolean flag
+	waitingForBeaconChain *uint32       // atomic boolean flag
+	skipCycleHack         chan struct{} // with this channel we tell the stagedsync that we want to assemble a block
+	miningResultPOSCh     <-chan *types.Block
+	assemblePayloadPOS    assemblePayloadPOSFunc
+	proposing             bool
+	pauseAssemble         *uint32
```

Comment:
I would make `pauseAssemble` a bool protected by `mu` for consistency with `pendingPayloads`